docs/StorageManagement.md

```diff
@@ -0,0 +1,235 @@
+    type: object
+    properties:
+      bucket:
+        type: string
+      endpoint:
+        type: string
+      object_key:
+        type: string
+      region:
+        type: string
+    required:
+    - bucket
+    - endpoint
+    - object_key
+```
+- Selection of modules is done based on connection `name`.
+
+In **Phase2** we add an optional `Category` field to `Connection`. 
+A module yaml will specify as a connection type either name or category. Optimizer will do the matching.
+
+### FybrikStorageAccount 
+Today, FybrikStorageAccount spec defines properties of s3 storage, e.g.:
+```
+spec:
+  id: theshire-object-store
+  secretRef: credentials-theshire
+  region: theshire
+  endpoint: http://s3.eu.cloud-object-storage.appdomain.cloud
+```
+
+We suggest to add `type` (connection name), `geography` and the appropriate connection properties.
+Example:
+```
+spec:
+  id: theshire-object-store
+  type: s3
+  secretRef: credentials-theshire
+  geography: theshire
+  s3:
+    region: eu
+    endpoint: http://s3.eu.cloud-object-storage.appdomain.cloud
+```
+
+Dynamic information about performance, amount free, costs, etc., are detailed in the separate Infrastructure Attributes JSON file.
+
+## StorageManager
+
+StorageManager is responsible for allocating storage in known storage accounts (as declared in FybrikStorageAccount CRDs) and for freeing the allocated storage.  
+
+
+
+### Architecture and interfaces
+
+StorageManager runs as a new container in the manager pod. A default Fybrik deployment uses its open-source implementation as a docker image specified in Fybrik `values.yaml`. This implementation can be replaced as long as the alternative obeys the following APIs:
+
+#### AllocateStorage
+
+Storage is allocated after the appropriate storage account has been selected by the optimizer. 
+
+`AllocateStorage` request includes properties of the selected storage account, asset name (and additional properties) defined in FybrikApplication, prefix for name generation based on application uuid, attributes defined by IT config policies, e.g., bucket_name. Upon a successful allocation, a connection object will be returned.
+
+#### DeleteStorage
+
+The allocated storage is freed after FybrikApplication is deleted or a dataset is no longer required in the spec, and the storage is not persistent.
+
+`DeleteStorage` request receives the `Connection` object to be deleted, and configuration options defined by IT config policies, e.g., delete_empty_bucket.
+
+#### GetSupportedConnectionTypes
+
+Returns a list of supported connection types. Optimizer will use this list to constrain selection of storage accounts. 
+
+
+## Architecture
+
+As the first step, storage management functionality will be defined in Fybrik repo under `pkg/storage`. 
+To consider moving to another repository in the future.
+
+The folder will include:
+
+- FybrikStorageAccount types to generate the CRD
+- Open-source implementation of StorageManager APIs
```

Comment:
... based on a some / all of the connection types in the taxonomy (_link to the section that discusses taxonomy_)

Reply:
OK